security/keys/encrypted-keys/encrypted.c

```diff
@@ -760,21 +841,25 @@ static int encrypted_init(struct encrypted_key_payload *epayload,
 	}
 
 	__ekey_init(epayload, format, master_desc, datalen);
-	if (!hex_encoded_iv) {
-		get_random_bytes(epayload->iv, ivsize);
-
-		get_random_bytes(epayload->decrypted_data,
-				 epayload->decrypted_datalen);
-	} else
-		ret = encrypted_key_decrypt(epayload, format, hex_encoded_iv);
-	return ret;
+  if (hex_encoded_iv) {
+    ret = encrypted_key_decrypt(epayload, format, hex_encoded_iv);
+  } else if (decrypted_data) {
+    get_random_bytes(epayload->iv, ivsize);
+    strcpy(epayload->decrypted_data, decrypted_data);
```

Comment:
strcpy is baaadd..
https://github.com/torvalds/linux/blob/master/Documentation/process/deprecated.rst

Reply:
I assume the locking here is good but double check

Reply:
modified to memcpy.